nativeruntime/cpp/CMakeLists.txt

```diff
@@ -13,6 +13,13 @@ set (CMAKE_CXX_STANDARD 17)
 
 project(nativeruntime)
 
+if (CMAKE_HOST_SYSTEM_NAME MATCHES "Windows")
```

Comment:
Maybe we can drop `find_package(JNI)`? That seems to cause problems on Windows, and it has caused problems on Mac before. Can you extract this chunk into a new PR and see if we can just add the include path and library?

Reply:
No problem. If we don't different build tools on Windows, the `CMAKE_HOST_SYSTEM_NAME` may change. It is not always `Windows` on Windows for different build tools.

Reply:
No problem.

Reply:
Emmm. This chunk needs `JAVA_HOME`, and use different `PATH2` for different platforms. Maybe I need check what CMake's `FindPackage(JNI)` does to support different platforms. I have saw some similar things at work, I can give a try for it.

Reply:
Looks like there are many fixes for `FindJNI.cmake` on newer cmake version: from commit 777e87ae at https://gitlab.kitware.com/cmake/cmake/-/commits/v3.22.0/Modules/FindJNI.cmake, especially commit https://gitlab.kitware.com/cmake/cmake/-/commit/696d16ae6c5214e314cfc7cb809c2e574bcff651, added aarch64 support. The lowest release version that contains those important commits is v3.18.4. If we can use this cmake version, maybe we can remove this trunk, otherwise we can copy `FindJNI.cmake` from cmake repository to Robolectric, and maintain it until we can bump cmake version.

Reply:
Looks like bumping cmake version is difficult. My Ubuntu 20.04 only has cmake 3.16.3.

Reply:
I created a PR to extract the patch that backports `FindJNI.cmake` at https://github.com/robolectric/robolectric/pull/6911.

Reply:
Related PR is here: https://github.com/robolectric/robolectric/pull/6914.